include/deal.II/base/mpi.h

```diff
@@ -1024,6 +1024,17 @@ namespace Utilities
                         const IndexSet &indices_to_look_up,
                         const MPI_Comm &comm);
 
+    /**
+     * Compute the union of the input vectors @p vec of all processes in the
+     *   MPI communicator @p comm.
+     *
+     * @note This is a collective operation. The result will available on all
+     *   processes.
+     */
+    template <typename T>
+    std::vector<T>
+    compute_union(const std::vector<T> &vec, const MPI_Comm &comm);
```

Comment:
Intuitively, I would expect that an `std::set` is returned. Is there any other reason than #9812 to choose `std::vector` instead?

Reply:
...or `std::unordered_set`.

Reply:
@masterleinad If you want, I can introduce second version of the function, which takes a set and returns a set. I would keep the vector version since the internal buffer in the vector is also a vector. Would that work with you?

Reply:
A lot of functions take in this namespace take an `ArrayView`. Maybe, it makes sense to do the same here (instead).

For me, returning a `std::vector` just feels a bit odd but is more compatible with `get_boundary_ids` (where we might have wanted to use an `std::set` in the first place). I don't feel that strongly about it, though.

Reply:
I agree -- the word "union" already somehow implies that the function operates on sets. Maybe change (or overload) it to take a `std::set` as argument and let it return that? I would also rename the function to `compute_set_union()` or simply `set_union()` to make it clear that that's what it is, rather than some kind of geometric union, for example.